source/AMDSThreadedTCPDataServer.cpp

```diff
@@ -0,0 +1,50 @@
+#include "source/AMDSThreadedTCPDataServer.h"
+
+#include <QCoreApplication>
+
+AMDSThreadedTCPDataServer::AMDSThreadedTCPDataServer(QString hwType, QObject *parent) :
+	QObject(parent)
+{
+	hwType_ = hwType;
+
+	thread_ = new QThread;
+	server_ = new AMDSTCPDataServer;
+
+	connect(thread_, SIGNAL(started()), this, SLOT(onThreadStarted()));
+	connect(this, SIGNAL(startServer(QString,quint16)), server_, SLOT(start(QString,quint16)));
+	connect(server_, SIGNAL(error(quint8,quint16,QString)), this, SIGNAL(error(quint8,quint16,QString)));
+
+	server_->moveToThread(thread_);
+	thread_->start(QThread::LowPriority);
+}
+
+AMDSThreadedTCPDataServer::~AMDSThreadedTCPDataServer()
+{
+	if (thread_->isRunning())
+		thread_->terminate();
```

Comment:
@iainworkman noticed this in a diffferent pull request, but you should probably be using `QThread::quit()` instead.